recipes/gcc/all/conanfile.py

```diff
@@ -1,126 +1,193 @@
-            "--libdir=%s" % libdir,
-            '--with-pkgversion=%s' % pkgversion,
-            "--program-suffix=-%s" % self.version,
-            "--with-bugurl=%s" % bugurl
-        ]
-        if self.settings.os == "Macos":
-            xcrun = tools.XCRun(self.settings)
-            args.extend([
-                '--with-native-system-header-dir=/usr/include',
-                "--with-sysroot={}".format(xcrun.sdk_path)
-            ])
-        self._autotools.libs = []  # otherwise causes config.log to fail finding -lmpc
+    def configure(self):
         if self.settings.compiler in ["clang", "apple-clang"]:
-            # xgcc: error: unrecognized command-line option -stdlib=libc++
-            if self.settings.compiler.libcxx == "libc++":
-                self._autotools.cxx_flags.remove("-stdlib=libc++")
-            elif self.settings.compiler.libcxx in ["libstdc++", "libstdc++11"]:
-                self._autotools.cxx_flags.remove("-stdlib=libstdc++")
-        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+            # Can't remove this from cxxflags with autotools - so get rid of it
+            del self.settings.compiler.libcxx
+
+    def build_requirements(self):
+        if self.info.settings.os == "Linux":
+            # binutils recipe is broken for Macos, and Windows uses tools
+            # distributed with msys/mingw
+            self.tool_requires("binutils/2.38")
+        self.tool_requires("flex/2.6.4")
 
     def requirements(self):
         self.requires("mpc/1.2.0")
         self.requires("mpfr/4.1.0")
-        self.requires("gmp/6.2.0")
-        self.requires("zlib/1.2.11")
+        self.requires("gmp/6.2.1")
+        self.requires("zlib/1.2.12")
+        self.requires("isl/0.24")
 
-    def configure(self):
-        if self.settings.os == "Windows":
-            raise ConanInvalidConfiguration("Windows builds aren't supported (yet), sorry")
-        if tools.cross_building(self.settings):
-            raise ConanInvalidConfiguration("no cross-building support (yet), sorry")
+    def package_id(self):
+        del self.info.settings.compiler
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = "gcc-%s" % self.version
-        os.rename(extracted_dir, self._source_subfolder)
+    def validate_build(self):
+        if self.settings.compiler in ["Visual Studio", "msvc"]:
+            raise ConanInvalidConfiguration("GCC can't be built with MSVC")
+
+    def validate(self):
+        if self.info.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "Windows builds aren't currently supported. Contributions to support this are welcome."
+            )
+        if self.info.settings.os == "Macos":
+            # This recipe should largely support Macos, however the following
```

Comment:
```suggestion
            # FIXME: This recipe should largely support Macos, however the following
```
Don't save keywords, they help us to find it again in the future.